profiles/entropia-cluster-vm/backup.nix

```diff
@@ -14,7 +14,18 @@
       BORG_RSH = "ssh -i /etc/ssh/ssh_host_ed25519_key";
       BORG_RELOCATED_REPO_ACCESS_IS_OK = "yes";
     };
+    preHook = lib.mkIf config.services.postgresql.enable ''
+      ${lib.getExe pkgs.sudo} -u postgres ${pkgs.postgresql}/bin/psql -t -A -c "SELECT datname FROM pg_database WHERE datname <> ALL ('{template0,template1,postgres}')" | ${pkgs.findutils}/bin/xargs -I DBNAME ${lib.getExe pkgs.sudo} -u postgres ${pkgs.postgresql}/bin/pg_dump -F directory -f /var/backup/postgresql/DBNAME DBNAME
```

Comment:
We should also run `pg_dumpall --globals-only` to backup cluster wide state. https://www.postgresql.org/docs/current/backup-dump.html#BACKUP-DUMP-ALL